gta/dff_importer.py

```diff
@@ -559,15 +566,20 @@ def import_frames():
                 elif frame.bone_data.header.id in self.bones and mesh is None:
                     continue
                     
-            
             # Create and link the object to the scene
             if obj is None:
                 obj = bpy.data.objects.new(frame.name, mesh)
                 link_object(obj, dff_importer.current_collection)
 
                 obj.rotation_mode       = 'QUATERNION'
-                obj.rotation_quaternion = matrix.to_quaternion()
-                obj.location            = frame.position
+                # obj.rotation_quaternion = matrix.to_quaternion()
+                # obj.location            = frame.position
+
+                # Dirty assignment
+                # Need to take existing location / rotation into account
+                obj.location = self.initial_object_position
+                obj.rotation_quaternion = self.initial_object_rotation
+                obj.scale = self.initial_object_scale
```

Comment:
This is a quick and dirty position and rotation assignment. I realise I'll have to add the existing position and rotation to my assigned values, now I'm just overwriting them.

But that aside, you originally suggested that:

> Then it should make new instances of the loaded models (possibly through Linked Duplicate function) and move them to the coordinates and other options set by the IPL file. This is done to not re-load the same file over and over again.

I could use some help with this, I don't really know what you meant by it or how to do it.
Or maybe what I have now is good enough?

The import process is quite slow, maybe your way of creating objects would speed things up a little?

Reply:
Yeah, could certainly improve with the optimisation in that. So now you're loading each IPL entry separately. This would mean that this would create duplicate meshes in memory, and worse, slow the importing process down. 

What I meant by that is, you can get a list of objects defined in an IDE files, load them, and store them in an array or dictionary. 
You can then create a new object(s) whenever it appears in the IPL file, and assign its mesh to the loaded one from the dictionary.

Something like this
```python
loaded_objects = {};

# While parsing the IDE
loaded_objects[ide.id] = dff_importer.import_dff(options);

# Later while parsing the IPL
bpy.data.objects.new("name", loaded_objects[ipl_entry.id]);
```

This way you're just creating another instance of the already loaded object, so you don't have to load it again. They won't share location or other object properties, but they will share the same mesh.
This will have these advantages:
- You won't have to load the same object again (this will save time and memory)
- Any changes you make to one mesh, will reflect the other so the scripts can (in the future) also act as useful map editors.

The export part will be a little tricky, but I reckon it would be possible to group the same objects in a collection or something similar.

You'd probably need to make the dff_importer return a list of objects or just get it from `dff_importer.objects`. Also something to keep in mind is, I think it is possible to have multiple objects in a single dff file?, so you'd probably have to loop over them if you use that method.